BLE_GAP/source/main.cpp

```diff
@@ -15,101 +15,81 @@
  */
 
 #include <events/mbed_events.h>
-#include <mbed.h>
 #include "ble/BLE.h"
-#include "gap/Gap.h"
-#include "gap/AdvertisingDataParser.h"
-#include "pretty_printer.h"
+#include "mbed-os-ble-utils/pretty_printer.h"
 
 /** This example demonstrates all the basic setup required
- *  to advertise, scan and connect to other devices.
+ *  to advertise and scan.
  *
  *  It contains a single class that performs both scans and advertisements.
  *
  *  The demonstrations happens in sequence, after each "mode" ends
- *  the demo jumps to the next mode to continue. There are several modes
- *  that show scanning and several showing advertising. These are configured
- *  according to the two arrays containing parameters. During scanning
- *  a connection will be made to a connectable device upon its discovery.
+ *  the demo jumps to the next mode to continue.
+ *
+ *  You may connect to the device during advertising and if you advertise
+ *  this demo will try to connect during the scanning phase. Connection
+ *  will terminate the phase early. At the end of the phase some stats
+ *  will be shown about the phase.
  */
 
-events::EventQueue event_queue;
-
-/* Duration of each mode in milliseconds */
-static const size_t MODE_DURATION_MS      = 6000;
-
-/* Time between each mode in milliseconds */
-static const size_t TIME_BETWEEN_MODES_MS = 2000;
-
-/* how long to wait before disconnecting in milliseconds */
-static const size_t CONNECTION_DURATION   = 3000;
+/* demo config */
+/* you can adjust these parameters and see the effect on the performance */
 
-/* how many advertising sets we want to crate at once */
-static const uint8_t ADV_SET_NUMBER       = 2;
+/* Advertising parameters are mainly defined by an advertising type and
+ * and an interval between advertisements. lower interval increases the
+ * chances of being seen at the cost of more power.
+ * The Bluetooth controller may run concurrent operations with the radio;
+ * to help it, a minimum and maximum advertising interval should be
+ * provided.
+ */
+static const ble::AdvertisingParameters advertising_params(
+    ble::advertising_type_t::CONNECTABLE_UNDIRECTED,
+    ble::adv_interval_t(40),
```

Comment:
It might be worth adding the time in ms in comments and/or mention the use of ble::millisecond_t.

Reply:
added